### PR TITLE
Fix that PPEffects Throw a error on HC

### PR DESCRIPTION
**Pull request by jokoho48**
Fix that PPEffects Throw a error on HC(this issue is PFH)

### CODE REVIEW

File: addons/common/functions/fnc_blurScreen.sqf [hunk 1]
```diff
@@ -10,6 +10,8 @@
 
 #include "script_component.hpp"
 
+if !(hasInterface) exitWith {};
```
Comment (bux578) [inline addons/common/functions/fnc_blurScreen.sqf]: I don't know who came up with the idea to move the exclamation mark outside of the parentheses, but it looks ugly and wrong to me.

`if (!hasInterface) exitWith {};`

Comment (jokoho48) [inline addons/common/functions/fnc_blurScreen.sqf]: ok hmm yeah i can move it in if you are happy with that

**Comment by jonpas** [comment]
This PR introduces a bug and a critical bug? XD

**Comment by jokoho48** [comment]
na its a bug fix bug a critical bug

File: addons/huntir/functions/fnc_cam.sqf [hunk 2]
```diff
@@ -13,6 +13,7 @@
  */
 #include "script_component.hpp"
 
+if !(hasInterface) exitWith {};
```
Comment (commy2) [inline addons/huntir/functions/fnc_cam.sqf]: I doubt this ever runs on a dedi/hc. Seems redundant.

File: addons/nightvision/functions/fnc_changeNVGBrightness.sqf [hunk 3]
```diff
@@ -16,6 +16,7 @@
  */
 #include "script_component.hpp"
 
+if !(hasInterface) exitWith {};
```
Comment (commy2) [inline addons/nightvision/functions/fnc_changeNVGBrightness.sqf]: Same. This is called by a CBA key handler. Just don't add the keyhandler.

File: addons/goggles/functions/fnc_onEachFrame.sqf [hunk 4]
```diff
@@ -14,6 +14,7 @@
  * Public: No
  */
 #include "script_component.hpp"
+if !(hasInterface) exitWith {};
```
Comment (commy2) [inline addons/goggles/functions/fnc_onEachFrame.sqf]: Don't add the PFH on a dedi/hc. Seems better than to check something every frame that will never change.

File: addons/goggles/functions/fnc_applyGlassesEffect.sqf [hunk 5]
```diff
@@ -16,6 +16,8 @@
  * Public: No
  */
 #include "script_component.hpp"
+if !(hasInterface) exitWith {};
```
Comment (commy2) [inline addons/goggles/functions/fnc_applyGlassesEffect.sqf]: >Don't add the PFH on a dedi/hc. Seems better than to check something every frame that will never change.

File: addons/goggles/XEH_postInit.sqf [hunk 6]
```diff
@@ -15,7 +15,7 @@
  * Public: No
  */
 #include "script_component.hpp"
-if (!hasInterface) exitWith {};
+if !(hasInterface) exitWith {};
```
Comment (commy2) [inline addons/goggles/XEH_postInit.sqf]: Doesn't his cover basically everything in goggles?

File: addons/gforces/functions/fnc_pfhUpdateGForces.sqf [hunk 7]
```diff
@@ -13,8 +13,10 @@
  */
  #include "script_component.hpp"
 
+if !(hasInterface) exitWith {};
```
Comment (commy2) [inline addons/gforces/functions/fnc_pfhUpdateGForces.sqf]: >Don't add the PFH on a dedi/hc. Seems better than to check something every frame that will never change.

**Comment by commy2** [comment]
I think most of these could be done in a smarter way. Don't call functions / add pfhs in the first place instead of checking if they don't apply to dedi/hc.

File: addons/common/functions/fnc_blurScreen.sqf (hunk 1 above)
Comment (MikeMatrix) [inline addons/common/functions/fnc_blurScreen.sqf]: I agree with @bux578 on this one. It makes absolutely no sense to have it outside.

Comment (SilentSpike) [inline addons/common/functions/fnc_blurScreen.sqf]: but...it makes more sense to have it outside! :laughing:

**Comment by Glowbal** [comment]
Status?

**Comment by jokoho48** [comment]
still need a review and testing but i think this work
and its importent for this version because this is a very critical bug

File: addons/spectator/functions/fnc_setSpectator.sqf [hunk 8]
```diff
@@ -23,7 +23,7 @@
 params [["_set",true,[true]]];
 
 // Only clients can be spectators
-if !(hasInterface) exitWith {};
+if (!hasInterface) exitWith {};
```
Comment (SilentSpike) [inline addons/spectator/functions/fnc_setSpectator.sqf]: What's with all these dirty edits being made in this PR?

**Pull request merged**